exporter/jaegerexporter/factory.go

```diff
@@ -0,0 +1,101 @@
+package jaegerexporter
+
+import (
+	"fmt"
+
+	"contrib.go.opencensus.io/exporter/jaeger"
+	"github.com/open-telemetry/opentelemetry-service/config/configerror"
+	"github.com/open-telemetry/opentelemetry-service/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-service/exporter"
+	"github.com/open-telemetry/opentelemetry-service/exporter/exporterwrapper"
+	"go.uber.org/zap"
+)
+
+var _ = exporter.RegisterFactory(&factory{})
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "jaeger"
+)
+
+// factory is the factory for Jaeger exporter.
+type factory struct {
+}
+
+// Type gets the type of the Exporter config created by this factory.
+func (f *factory) Type() string {
+	return typeStr
+}
+
+// CreateDefaultConfig creates the default configuration for exporter.
+func (f *factory) CreateDefaultConfig() configmodels.Exporter {
+	return &ConfigV2{
+		ExporterSettings: configmodels.ExporterSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr,
+		},
+	}
+}
+
+// CreateTraceExporter creates a trace exporter based on this config.
+func (f *factory) CreateTraceExporter(logger *zap.Logger, config configmodels.Exporter) (consumer.TraceConsumer, exporter.StopFunc, error) {
+	jc := config.(*ConfigV2)
+
+	if jc.CollectorEndpoint == "" {
+		return nil, nil, &jTraceExporterError{
+			code: errCollectorEndpointRequired,
+			msg:  "Jaeger exporter config requires an Endpoint",
+		}
+	}
+
+	jOptions := jaeger.Options{}
+	jOptions.CollectorEndpoint = jc.CollectorEndpoint
+
+	if jc.Username == "" {
+		return nil, nil, &jTraceExporterError{
+			code: errUsernameRequired,
+			msg:  "Jaeger exporter config requires a Username",
+		}
+	}
+	jOptions.Username = jc.Username
+
+	if jc.Password == "" {
+		return nil, nil, &jTraceExporterError{
+			code: errPasswordRequired,
+			msg:  "Jaeger exporter config requires a Password",
+		}
+	}
+	jOptions.Password = jc.Password
+
+	if jc.Password == "" {
+		return nil, nil, &jTraceExporterError{
+			code: errServiceNameRequired,
+			msg:  "Jaeger exporter config requires a ServiceName",
+		}
+	}
+	jOptions.Process = jaeger.Process{
+		ServiceName: jc.ServiceName,
+	}
+
+	exporter, serr := jaeger.NewExporter(jOptions)
+	if serr != nil {
+		return nil, nil, fmt.Errorf("cannot configure jaeger Trace exporter: %v", serr)
+	}
+
+	jexp, err := exporterwrapper.NewExporterWrapper("jaeger", "ocservice.exporter.Jaeger.ConsumeTraceData", exporter)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return jexp, noopStopFunc, nil
```

Comment:
Stop function is needed for Jaeger. It should not be noop. It needs to call exporter.Flush(). See current implementation in jaeger.go/JaegerExportersFromViper.

Reply:
I attempted the same implementation [je.stop in a previous commit](https://github.com/open-telemetry/opentelemetry-service/pull/136/commits/61d161873756aee21735e39263d2bdcae9679b7b#diff-dab60d735cdff673a25d6a2ab3bed2b8R107) following ["exporter.Flush()"](https://godoc.org/contrib.go.opencensus.io/exporter/jaeger#Exporter.Flush) but I thought maybe I would try to do nothing and It will be reviewed anyway.